Frontend/ui-driver/src/Screens/DriverEarningsScreen/Controller.purs

```diff
@@ -0,0 +1,465 @@
+                    "CANCELLED" -> Just 0
+                    _ -> ride.computedFare,
+  status : Just ride.status, 
+  coins :  0,
+  event : "",
+  tagImages : getTagImages (RidesInfo ride),
+  cash : 0.0
+}) list) 
+
+getTagImages :: RidesInfo -> Array String
+getTagImages (RidesInfo ride) = do
+  let a = maybe [] (\_ -> ["ny_ic_tip_ride_tag"]) ride.customerExtraFee
+      b = maybe [] (\_ -> ["ny_ic_disability_tag"]) ride.disabilityTag
+      c = maybe [] (\_ -> ["ny_ic_special_location_tag"]) ride.specialLocationTag
+      d = maybe [] (\_ -> ["ny_ic_goto_home_tag"]) ride.driverGoHomeRequestId
+  (a <> b <> c <> d)
+
+getTotalCurrentWeekData :: Array ST.WeeklyEarning -> ST.TotalEarningsData
+getTotalCurrentWeekData barGraphData = do
+  let firstElement = barGraphData !! 0
+      lastElement = barGraphData !! 6
+      calculateTotals = foldl (\acc record ->
+        { totalEarnings: acc.totalEarnings + record.earnings
+        , totalDistance: acc.totalDistance + record.rideDistance
+        , totalRides: acc.totalRides + record.noOfRides
+        }) { totalEarnings: 0, totalDistance: 0, totalRides: 0 } barGraphData
+  { fromDate : maybe "" (\x -> x.rideDate) firstElement,
+    toDate : maybe "" (\x -> x.rideDate) lastElement,
+    totalEarnings : calculateTotals.totalEarnings,
+    totalRides : calculateTotals.totalRides,
+    totalDistanceTravelled : calculateTotals.totalDistance
+  }  
+
+fetchWeekyEarningData :: KeyStore -> Maybe (Array ST.WeeklyEarning)
+fetchWeekyEarningData name = do
+ let result = decodeJSON $ getValueToLocalNativeStore name
+ case (runExcept $ result) of
+  Left err -> Nothing
+  Right (ST.CachedEarningsForDriver weeklyEarnings) -> if weeklyEarnings.id == getValueToLocalStore DRIVER_ID then Just weeklyEarnings.earningsData else Nothing
+
+getEarningDataOfWeek :: Int -> Array ST.WeeklyEarning -> Array ST.WeeklyEarning
+getEarningDataOfWeek index weeklyEarningData = 
+  case index of
+    3 -> DA.drop 21 weeklyEarningData
+    2 -> DA.take 7 (DA.drop 14 weeklyEarningData)
+    1 -> DA.take 7 (DA.drop 7 weeklyEarningData)
+    0 -> DA.take 7 weeklyEarningData
+    _ -> []
+
+getWeeklyEarningsPercentage :: Array ST.WeeklyEarning -> Array ST.WeeklyEarning
+getWeeklyEarningsPercentage weeklyEarningData = map (\x-> x{percentLength = ((toNumber x.earnings) * 100.0)/ (toNumber maxValue)}) weeklyEarningData
+  where maxValue = foldl getMax 0 weeklyEarningData
+
+getMax :: Int -> ST.WeeklyEarning -> Int
+getMax num1 obj1 = if obj1.earnings > num1 then obj1.earnings else num1
+
+updateWeeklyEarningData :: Array ST.WeeklyEarning -> Array ST.WeeklyEarning -> Int -> Array ST.WeeklyEarning
+updateWeeklyEarningData currWeekData allWeeksData weekIndex = do
+  let obj = {
+    earnings : 0,
+        rideDistance : 0,
+        rideDate : "",
+        noOfRides : 0,
+        percentLength : 0.0
+  }
+  let fillData = DA.replicate (weekIndex * 7) obj
+  case weekIndex of 
+    3 ->  fillData <> currWeekData <> DA.drop 28 allWeeksData
```

Comment:
instead of using case match we can do `fillData <> currWeekData <> DA.drop ((weekIndex+1)*7) allWeeksData` instead.